script/server

```diff
@@ -65,4 +65,32 @@ main()
     . "$AFTER_HOOK"
 }
 
-main
+shutdown()
+{
+    nat64_stop || echo 'Failed to start NAT64!'
+    dns64_stop || echo 'Failed to start DNS64!'
+    firewall_stop || echo 'Failed to start firewall'
+    if have systemctl; then
+        systemctl is-active rsyslog && sudo systemctl stop rsyslog || echo 'Failed to stop rsyslog!'
+        systemctl is-active dbus && sudo systemctl stop dbus || echo 'Failed to stop dbus!'
+        systemctl is-active avahi-daemon && sudo systemctl stop avahi-daemon || echo 'Failed to stop avahi!'
+        without WEB_GUI || systemctl is-active otbr-web && sudo systemctl stop otbr-web || echo 'Failed to stop otbr-web!'
+        systemctl is-active otbr-agent && sudo systemctl stop otbr-agent || echo 'Failed to stop otbr-agent!'
+    elif have service; then
+        sudo service rsyslog status && sudo service rsyslog stop || echo 'Failed to stop rsyslog!'
+        sudo service dbus status && sudo service dbus stop || echo 'Failed to stop dbus!'
+        sudo service mdns status && sudo service mdns stop || echo "service mdns is not available!"
+        sudo service avahi-daemon status && sudo service avahi-daemon stop || echo 'Failed to stop avahi!'
+        sudo service otbr-agent status && sudo service otbr-agent stop || echo 'Failed to stop otbr-agent!'
+        without WEB_GUI || sudo service otbr-web status && sudo service otbr-web stop || echo 'Failed to stop otbr-web!'
+    else
+        echo 'Unable to find service manager. Try script/console to stop in console mode!'
+    fi
+}
+
+if [ "${1-}" == "shutdown" ]
+then
+    shutdown
+else
+    main
+fi
```

Comment:
prefer main function as the single entry point for a script. How about rename the current main() to startup() and wrap these lines as the new main()?
see more: https://google.github.io/styleguide/shellguide.html#main
```suggestion
main()
{
    if [[ "${1-}" == "shutdown" ]]; then
        shift
        shutdown "$@"
    else
        startup "$@"
    fi
}
```

Reply:
I think it's good now.